### PR TITLE
Updating Hashing function to Murmur3

### PR DESCRIPTION
**Pull request by HJewkes**

### CODE REVIEW

File: src/main/java/io/split/client/utils/MurmurHash3.java [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+    /**
+     * Returns the MurmurHash3_x86_32 hash.
+     */
+    public static int murmurhash3_x86_32(byte[] data, int offset, int len, int seed) {
+
+        final int c1 = 0xcc9e2d51;
+        final int c2 = 0x1b873593;
+
+        int h1 = seed;
```
Comment (patricioe) [inline src/main/java/io/split/client/utils/MurmurHash3.java]: @adilaijaz the seed should be used at the very end correct?

**Comment by adilaijaz** [comment]
Pato, where the seed gets used depends on the hashing algorithm. In murmur,
it gets used in the beginning.

File: client/src/main/java/io/split/engine/splitter/Splitter.java [hunk 2]
```diff
@@ -11,8 +12,10 @@
  * @author adil
  */
 public class Splitter {
+    private static final int ALGO_MURMUR = 1;
```
Comment (patricioe) [inline client/src/main/java/io/split/engine/splitter/Splitter.java]: murmur is actually #2. see ticket: https://splitio.atlassian.net/browse/SPLIT-2578

Comment (patricioe) [inline client/src/main/java/io/split/engine/splitter/Splitter.java]: yeah confirmed. This is not correct.

**Comment by HJewkes** [comment]
LGTM

**Pull request merged**